modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/loadData.ts

```diff
@@ -180,6 +182,37 @@ export async function loadData({
 
 				url.searchParams.set('nestedFields', newNestedFields.join(','));
 			}
+			else if (
+				key === 'nestedFieldsDepth' &&
+				existingNestedFieldsDepth
+			) {
+				const existingNestedFieldsDepthValue = Number(
+					existingNestedFieldsDepth
+				);
+				const additionalAPIURLParametersNestedFieldsDepthValue =
+					Number(value);
+
+				if (
+					!isNaN(existingNestedFieldsDepthValue) &&
+					!isNaN(additionalAPIURLParametersNestedFieldsDepthValue)
+				) {
+					url.searchParams.set(
+						key,
+						existingNestedFieldsDepthValue >=
+							additionalAPIURLParametersNestedFieldsDepthValue
+							? existingNestedFieldsDepth
+							: value
```

Comment:
in case we have 2 `nestedFieldsDepth` values (one set by custom serializer in the API URL, and the other coming via additional url parameters), we use the bigger value.